awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -88,7 +88,28 @@ def get_all_endpoint(self, endpoint, *args, **kwargs):
             response['json']['next'] = next_page
         return response
 
-    def get_one(self, endpoint, *args, **kwargs):
+    def get_name_field_from_endpoint(self, endpoint):
+        if endpoint == 'users':
+            return 'username'
+        elif endpoint == 'instances':
+            return 'hostname'
+        return 'name'
```

Comment:
You're cycling over potential identity fields, but that's the same thing done in other cases like this:

https://github.com/ansible/awx/blob/c4393c4827691139c473b3eaa0a05c2a2b4e6aac/awx_collection/plugins/module_utils/tower_api.py#L349-L360

Your implementation manages to reuse `get_name_field_from_endpoint` in 2 places, but this logic is still very fractured and duplicated.

My thoughts on implementation here is that you probably want a global OrderedDict or simple list of tuples so that you can cycle over them in both cases. That way, you can map the field name exceptions to their endpoints, users-username, instances-hostname, workflow_job_template_nodes-identifier.

I really want this helper method to be totally comprehensive, in a way that will cause it to get used over and over again. Probably a lot. You could have a base method that returns the field name (like username) and wrap that with another method that returns the value (like joe).